sdk/typescript/src/client/http-transport.ts

```diff
@@ -53,20 +53,20 @@ export class SuiHTTPTransport implements SuiTransport {
 	}
 
 	fetch(input: RequestInfo, init?: RequestInit): Promise<Response> {
-		const fetch = this.#options.fetch ?? globalThis.fetch;
+		const fetchFn = this.#options.fetch ?? fetch;
 
-		if (!this.fetch) {
+		if (!fetchFn) {
 			throw new Error(
 				'The current environment does not support fetch, you can provide a fetch implementation in the options for SuiHTTPTransport.',
 			);
 		}
 
-		return fetch(input, init);
+		return fetchFn(input, init);
 	}
 
 	#getWebsocketClient(): WebsocketClient {
 		if (!this.#websocketClient) {
-			const WebSocketConstructor = this.#options.WebSocketConstructor ?? globalThis.WebSocket;
```

Comment:
There have been some reports of this not working correctly, not sure why that would be, but I am hoping this change will improve it a bit